Docs 6444 3.9 connector docs (#971)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/microsoft-service-bus-connector.adoc

= Microsoft Service Bus Connector - Mule 3
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> 

Reviewer: EDIT INSTRUCTIONS

verify this PR: https://github.com/mulesoft/docs-mule-runtime/pull/971 — Green only please

Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#anypoint-connectors[Select]

Microsoft Service Bus Connector v1.3.n

Anypoint Connector for Microsoft Service Bus enables message integration with Windows Service Bus on-premises and Azure Service Bus on the cloud. The connector supports communication with queues, topics, and event hubs through AMQP 1.0. In addition, dynamic discovery and provisioning of Service Bus objects is possible using the built-in management API.

Release Notes: xref:release-notes::connector/microsoft-service-bus-connector-release-notes.adoc[Microsoft Service Bus Connector Release Notes] +
Reference: https://mulesoft.github.io/mule-microsoft-service-bus-connector/[Microsoft Service Bus Connector Reference]

The connector currently supports the following operations:

> * Send to queues and topics with support of AMQP message properties and header, including custom properties.

Reviewer: ```suggestion
* Send to queues and topics with support of AMQP message properties and header, including custom properties
```
List items not sentences

> * Receive from Queues and Topics asynchronously

Reviewer: Lowercase queues and topics

== Prerequisites

> This document assumes that you are familiar with Microsoft Service Bus, Mule, xref:anypoint-connectors.adoc[Anypoint Connectors], and the xref:6@studio::index.adoc[Anypoint Studio]. To increase your familiarity with Studio, consider completing one or more xref:6@studio::basic-studio-tutorial.adoc[Anypoint Studio Tutorials]. Further, this page assumes that you have a basic understanding of xref:mule-concepts.adoc[Mule flows] and xref:global-elements.adoc[Mule Global Elements].

Reviewer: GLOBAL TEMPLATE EDIT
```suggestion
You are familiar with Microsoft Service Bus, Mule runtime engine, xref:anypoint-connectors.adoc[Anypoint Connectors], and the xref:6@studio::index.adoc[Anypoint Studio]. To increase your familiarity with Studio, consider completing one or more xref:6@studio::basic-studio-tutorial.adoc[Anypoint Studio Tutorials]. Further,  you have a basic understanding of xref:mule-concepts.adoc[Mule flows] and xref:global-elements.adoc[Mule Global Elements].
```

Reviewer: Reword to not use "this document assumes."

> == To Install this Connector

Reviewer: ```suggestion
== To Install This Connector
```

. In Anypoint Studio, click the Exchange icon in the Studio taskbar.

> . Click *Login* in Anypoint Exchange.

Reviewer: I wonder if we should GLOBALLY replace this with "1. Log in to Anypoint Exchange" or "1. From Anypoint Exchange, search for the connector and click *Install*"?

Just a thought I keep having....

|===
|Field |Description
|*Name* |Enter a name for the global element.
|*Service Namespace* |Enter the name of the service namespace to address Service Bus resources within your app.

> |such as, Topic level Shared Key) will not work with this option, unless you disable connectivity test at startup. See the *Skip Connectivity Test* option.

Reviewer: MISSING TEXT

Reviewer: MISSING TEXT plus...
```suggestion
|such as a topic-level shared Key) does not work with this option, unless you disable connectivity test at startup. See the *Skip Connectivity Test* option.
```
Are we instructing the audience to see a UI label, or are we xref'ing a text section? Clarify.

|*Connector Configuration* |Microsoft_Service_Bus_Azure_Service_Bus (Refer to Step 2 to learn how to create a global element)
|*Operation* |Queues List
|===
+
. Drag an Object to JSON transformer next to the Microsoft Service Bus connector.

=== Run the App

> You are now ready to run the project! First, you can test run the app from Studio:

Reviewer: ```suggestion
Before you run the project, test run the app from Studio:
```

To enable the SSL checks, the certificate must be imported following these steps:

> . Use the PowerShell command https://msdn.microsoft.com/library/azure/jj248762%28v=azure.10%29.aspx[Get-SBAutoGeneratedCA] to download the certificate locally in the box running the Windows Service Bus. For the purposes of this tutorial, assume the certificate file is exported to `%temp%\AutoGeneratedCA.cer`.

Reviewer: ```suggestion
. Use the PowerShell command https://msdn.microsoft.com/library/azure/jj248762%28v=azure.10%29.aspx[Get-SBAutoGeneratedCA] to download the certificate locally in the box running the Windows Service Bus. 
For the purposes of this tutorial, assume the certificate file is exported to `%temp%\AutoGeneratedCA.cer`.
```

> . Go to `%programfiles%\Java\jre7`. Verify that the `bin\keytool.exe` tool exists, and that `lib\security\cacerts` exists. Note that you must be running as Administrator in order to perform a certificate import with Keytool.exe. Otherwise, an Access Denied error is generated.

Reviewer: ```suggestion
. Go to `%programfiles%\Java\jre7` to verify that both the `bin\keytool.exe` tool and `lib\security\cacerts` exist.
Note that you must be running as Administrator to perform a certificate import with Keytool.exe. Otherwise, an Access Denied error is generated.
```
This step is verification. How does that relate to importing a certificate? Do these two sentences that follow the step really belong here? Should the step just begin with "As an Administrator user, go to..."?

As the SAS token schema is URI based (that is, you can assign different authorization access to your resources based on their URIs) the connection supports multiple ways of providing the authentication tokens needed. The most trivial and simple case is when you have a single profile that authorizes accesses to all your resources by using an specific root URI which is the base endpoint that your service expose. If you need to provide different access tokens on different resources, then you can use a setting that allows you to configure a list of them according to the resources the connector will need to access during its running time.

> Last but not least, and extending the mechanisms to provide the authentication token, there is another mechanism that allows you to implement a custom token provider to allow the connector requests for security tokens when needed. It is your responsibility to solve each request and to return a fresh token every time the connector asks for one. As the token has an expiration time within it, this mechanism allows the connector to re-authenticate with the target resource once the token has expired (this is not allowed with the previous mechanisms where the tokens are fixed at configuration time before the flow runs).

Reviewer: ```suggestion
Also, when you solve each request and return a fresh token every time the connector asks for one, you allow the connector to reauthenticate with the target resource after the token expires.
```

|*Subscription* |In case of receiving messages from a topic, the name of the subscription from where the messages is retrieved has to be specified
|===

> To use these operations, the connector has to be an inbound endpoint. The `Receive` operations use asynchronous listeners to receive the messages. Once the message is received, the custom properties of the AMQP Message is transformed into Mule Message properties with the “amqp.” prefix, and the content of the message is passed as the payload.

Reviewer: ```suggestion
To use these operations, the connector has to be an inbound endpoint. The `Receive` operations use asynchronous listeners to receive the messages. After the message is received, the custom properties of the AMQP message are transformed into Mule message properties with the `amqp` prefix, and the content of the message is passed as the payload.
```
